packages/components/src/InputFile/InputFile.tsx

```diff
@@ -224,8 +229,14 @@ export function InputFile({
       httpMethod = "POST",
     } = await getUploadParams(file);
 
-    const fileUpload = getFileUpload(file, key);
-    onUploadStart && onUploadStart({ ...fileUpload });
+    const fileUpload = {
+      ...getFileUpload(file, key),
+      uploadUrl: url || undefined,
```

Comment:
I was thinking more on the lines of

```
{
  ...
  ...(url && { uploadUrl: url })
}
```

That's the reason why the test changed. Since it will be there now wether it's undefined or nah.

but if the answer above is "this should always be there" then don't need to do this.

Reply:
remove the `|| undefined` since it wont be undefined now